scss/_patterns_accordion.scss

```diff
@@ -59,53 +61,93 @@
-    padding-top: 0;
-    position: relative;
-    text-indent: -$sph-inner - $icon-size;
+  @include deprecate('3.0.0', '.p-accordion__title and .p-accordion__tab--with-title are being deprecated, please use .p-accordion__heading > .p-accordion__tab pattern instead') {
+    .p-accordion__title {
+      margin-bottom: 0;
+      padding-top: 0;
+      position: relative;
+      text-indent: -$sph-inner - $icon-size;
+
+      &::before {
+        background-size: $icon-size;
+        content: '';
+        display: inline-block;
+        height: $icon-size;
+        margin-right: $sph-inner;
+        width: $icon-size;
+      }
+    }
 
-    &::before {
-      background-size: $icon-size;
-      content: '';
-      display: inline-block;
-      height: $icon-size;
-      margin-right: $sph-inner;
-      width: $icon-size;
+    // FIXME: we shouldn't be targeting types directly here
+    // stylelint-disable selector-max-type
+    h2.p-accordion__title::before {
+      // make icon bigger for h2 (same as p-icon)
+      $icon-size-h2: $x-height;
+
+      background-size: $icon-size-h2;
+      height: $icon-size-h2;
+      width: $icon-size-h2;
     }
-  }
 
-  // FIXME: we shouldn't be targeting types directly here
-  // stylelint-disable selector-max-type
-  h2.p-accordion__title::before {
-    // make icon bigger for h2 (same as p-icon)
-    $icon-size-h2: $x-height;
+    h5.p-accordion__title::before,
+    h6.p-accordion__title::before {
+      // align icon within font metrics same as p-icon pattern
+      $vertical-offset: 0.5px;
 
-    background-size: $icon-size-h2;
-    height: $icon-size-h2;
-    width: $icon-size-h2;
-  }
+      vertical-align: calc(#{$vertical-offset} + #{0.5 * $cap-height} - #{0.5 * $icon-size});
+    }
+    // stylelint-enable selector-max-type
 
-  h5.p-accordion__title::before,
-  h6.p-accordion__title::before {
-    // align icon within font metrics same as p-icon pattern
-    $vertical-offset: 0.5px;
+    // override base expanded button styles
+    .p-accordion__tab--with-title[aria-expanded] {
+      background-color: inherit;
+    }
 
-    vertical-align: calc(#{$vertical-offset} + #{0.5 * $cap-height} - #{0.5 * $icon-size});
-  }
-  // stylelint-enable selector-max-type
+    .p-accordion__tab--with-title[aria-expanded='true'] .p-accordion__title::before {
+      @include vf-icon-minus($color-mid-dark);
+    }
 
-  // override base expanded button styles
-  .p-accordion__tab--with-title[aria-expanded] {
-    background-color: inherit;
+    .p-accordion__tab--with-title[aria-expanded='false'] .p-accordion__title::before {
+      @include vf-icon-plus($color-mid-dark);
+    }
   }
 
-  .p-accordion__tab--with-title[aria-expanded='true'] .p-accordion__title::before {
-    @include vf-icon-minus($color-mid-dark);
-  }
+  .p-accordion__heading {
+    margin-bottom: 0;
+    max-width: none;
+    padding-top: 0;
+
+    @at-root {
+      h2#{&} > .p-accordion__tab,
+      &.p-heading--two > .p-accordion__tab,
```

Comment:
`--two`, etc class names are deprecated. We don't need to add support for them in new code. If someone is about to use new accordion structure, they should use proper heading elements or class names.